src/logPointsManager.ts

```diff
@@ -0,0 +1,420 @@
+    startSession(site: WebSiteModels.Site, affinityValue: string, publishCredential: WebSiteModels.User): Promise<CommandRunResult<StartSessionResponse>> {
+        return this.makeCallAndLogException<StartSessionResponse>(site, affinityValue, publishCredential, "curl -X POST http://localhost:32923/debugger/session");
+    }
+
+    enumerateProcesses(site: WebSiteModels.Site, affinityValue: string, publishCredential: WebSiteModels.User): Promise<CommandRunResult<EnumerateProcessResponse>> {
+        return this.makeCallAndLogException<EnumerateProcessResponse>(site, affinityValue, publishCredential, "curl -X GET http://localhost:32923/os/processes?applicationType=Node.js");
+    }
+
+    attachProcess(site: WebSiteModels.Site, affinityValue: string, publishCredential: WebSiteModels.User, data: AttachProcessRequest): Promise<CommandRunResult<AttachProcessResponse>> {
+        return this.makeCallAndLogException<AttachProcessResponse>(site, affinityValue, publishCredential,
+            `curl -X POST -H "Content-Type: application/json" -d '{"processId":"${data.processId}","codeType":"javascript"}' http://localhost:32923/debugger/session/${data.processId}/debugee`);
+    }
+
+    call<ResponseType>(site: WebSiteModels.Site, affinityValue: string, publishCredential: WebSiteModels.User, command: string): Promise<CommandRunResult<ResponseType>> {
+        site && affinityValue && publishCredential;
+
+        return new Promise<CommandRunResult<ResponseType>>((resolve) => {
+            child_process.exec(command, (error: any, stdout: string, stderr: string) => {
+                if (error) {
+                    resolve(new CommandRunResult<ResponseType>(error, error.code, stderr));
+                    return;
+                }
+
+                resolve(new CommandRunResult<ResponseType>(null, 0, stdout));
+            });
+        });
+    }
+}
+
+let logPointsDebuggerClient: LogPointsDebuggerClient;
+
+if (shouldUseMockKuduCall) {
+    logPointsDebuggerClient = new MockLogpointsDebuggerClient();
+} else {
+    logPointsDebuggerClient = new KuduLogPointsDebuggerClient();
+}
+
+class CommandRunResult<ResponseType extends { error?: any, data?: any }> {
+    private _json: ResponseType;
+    constructor(public error: any, public exitCode: number, public output: string) {
+        this._json = undefined;
+    }
+
+    isSuccessful() {
+        return this.exitCode === 0 && this.json && !this.json.error;
+    }
+
+    get json(): ResponseType {
+        if (this._json === undefined) {
+            try {
+                this._json = JSON.parse(this.output);
+            } catch (err) {
+                util.getOutputChannel().appendLine(`API call error ${err.toString()}`);
+                this._json == null;
+            }
+        }
+
+        return this._json;
+    };
+}
+
+export class LogPointsSessionAttach extends WizardBase {
+    private _cachedPublishCredential: WebSiteModels.User
+
+    readonly hasSlot: boolean;

```

Comment:
We were debating whether we want to add "public" qualifiers to the public fields, since we saw that the existing code does not seem to do so. 

Please suggest.

Reply:
We are working on enabling tslint, and once that is in, "public" will be required by tslint.

Reply:
Addressed in https://github.com/Microsoft/vscode-azureappservice/pull/177 instead.